### ISSUE TITLE
  SubtaskStatus.starting: TypeError: 'SubtaskInstance' object is not subscriptable; Provider can't compute subtask: REASON.OfferCancelled

### ISSUE THREAD

**Issue by ederenn** [golemfactory/clay#4915]
## Description

**Golem Version**: 2019-11-19 15:51:01 INFO golemapp GOLEM Version: 0.21.0+dev460.g0c9ad08

**Golem-Messages version** (leave empty if unsure): 3.14.0

**Electron version** (if used): 0.21.0

**OS** [e.g. Windows 10 Pro]: All

**Branch** (if launched from source): b0.22-multicore-wasm

**Mainnet/Testnet**:

**Priority label is set to the lowest by default. To setup higher priority please change the label**
_P0 label is set for Severity-Critical/Effort-easy
P1 label is set  for Severity-Critical/Effort-hard
P2 label is set for Severity-Low/ Effort-easy
P3 label is set for Severity-Low/Effort-hard_

**Description of the issue**:

When subtask offer is cancelled for provider, it gives critical in logs,  `SubtaskStatus.starting: TypeError: 'SubtaskInstance' object is not subscriptable` `Provider can't compute subtask: REASON.OfferCancelled` this stops tasks from finishing. Subtasks ends in failure, provider is blocked.
 
**Actual result**:

_What is the observed behavior and/or result in this issue_

**Screenshots**:

```
2019-11-20 14:13:21 INFO     golem.task.tasksession              Provider can't compute subtask: '859f5a14-0b97-11ea-a9ae-2f67ff27ad1e' Reason: <REASON.OfferCancelled: 'OfferCancelled'>
2019-11-20 14:13:21 CRITICAL twisted                             
Traceback (most recent call last):
  File "/home/ederenn/projects/golem-env/lib/python3.6/site-packages/twisted/internet/asyncioreactor.py", line 136, in _readOrWrite
    why = method()
  File "/home/ederenn/projects/golem-env/lib/python3.6/site-packages/twisted/internet/tcp.py", line 243, in doRead
    return self._dataReceived(data)
  File "/home/ederenn/projects/golem-env/lib/python3.6/site-packages/twisted/internet/tcp.py", line 249, in _dataReceived
    rval = self.protocol.dataReceived(data)
  File "/home/ederenn/projects/golem/golem/network/transport/tcpnetwork.py", line 326, in dataReceived
    self._interpret(data)
  File "/home/ederenn/projects/golem/golem/network/transport/tcpnetwork.py", line 352, in _interpret
    self.session.interpret(m)
  File "/home/ederenn/projects/golem/golem/task/tasksession.py", line 196, in interpret
    BasicSafeSession.interpret(self, msg)
  File "/home/ederenn/projects/golem/golem/network/transport/session.py", line 75, in interpret
    action(msg)
  File "/home/ederenn/projects/golem/golem/task/tasksession.py", line 725, in _react_to_cannot_compute_task
    timeout,
  File "/home/ederenn/projects/golem/golem/core/common.py", line 179, in func_wrapper
    return func(*args, **kwargs)
  File "/home/ederenn/projects/golem/golem/task/taskmanager.py", line 860, in task_computation_cancelled
    new_status=SubtaskStatus.cancelled,
  File "/home/ederenn/projects/golem/golem/core/common.py", line 179, in func_wrapper
    return func(*args, **kwargs)
  File "/home/ederenn/projects/golem/golem/task/taskmanager.py", line 983, in restart_subtask
    self.tasks[task_id].restart_subtask(subtask_id)
  File "/home/ederenn/projects/golem/apps/wasm/task.py", line 548, in restart_subtask
    vbr_subtask.restart_subtask(subtask_id)
  File "/home/ederenn/projects/golem/apps/wasm/task.py", line 149, in restart_subtask
    if subtask['status'] != SubtaskStatus.starting:
TypeError: 'SubtaskInstance' object is not subscriptable

```

## Steps To Reproduce
_Short description of steps to reproduce the behavior:_
e.g.
1. Launch Golem on two nodes
2. Request simultaneously a dask on different requestor, best one Gwasm, one Blender 
3. Wait for computations to start, check if nodes are failing
4. See error in `golem.log`

## Expected behavior
_(What is the expected behavior and/or result in this scenario)_

## Logs and any additional context
_(Any additional information that could help to resolve the issue, which systems were checked to reproduce the issue)_
_Please upload your logs if possible_

## Proposed Solution?
_(Optional: What could be a solution for that issue)_

**Comment by prekucki** [comment]
bug is also on b0.22 and on mainet (b0.21).

**Comment by kmazurek** [comment]
Closed by #4933
